src/integrationTest/java/uk/nhs/hee/tis/trainee/forms/service/PdfServiceIntegrationTest.java

```diff
@@ -0,0 +1,209 @@
+        .lastModified(LocalDate.of(2026, 7, 8).atTime(9, 10).atZone(zoneId).toInstant())
+        .status(StatusDto.builder()
+            .current(StatusInfoDto.builder()
+                .state(LifecycleState.SUBMITTED)
+                .build())
+            .build())
+        .personalDetails(PersonalDetailsDto.builder()
+            .title("Dr")
+            .forenames("Anthony")
+            .surname("Gilliam")
+            .email("anthony.gilliam@example.com")
+            .telephoneNumber("07700 900000")
+            .mobileNumber("07700 900000")
+            .gmcNumber("1234567")
+            .gdcNumber("D123456")
+            .skilledWorkerVisaHolder(false)
+            .build())
+        .programmeMembership(ProgrammeMembershipDto.builder()
+            .name("General Practice")
+            .startDate(LocalDate.of(2024, 3, 1))
+            .endDate(LocalDate.of(2027, 3, 1))
+            .wte(0.8)
+            .build())
+        .change(CctChangeDto.builder()
+            .wte(0.6)
+            .startDate(LocalDate.of(2026, 3, 1))
+            .endDate(LocalDate.of(2027, 3, 1))
+            .cctDate(LocalDate.of(2028, 3, 1))
+            .build())
+        .reasons(ReasonsDto.builder()
+            .selected(List.of("Caring Responsibilities", "Other"))
+            .otherDetail("A trainee described reason detail")
+            .build())
+        .discussions(DiscussionsDto.builder()
+            .tpdName("Tee Pee-Dee")
+            .tpdEmail("tpd@example.com")
+            .other(List.of(
+                PersonDto.builder()
+                    .name("Ed Super")
+                    .email("ed.super@example.com")
+                    .role("Educational Supervisor")
+                    .build(),
+                PersonDto.builder()
+                    .name("Person Two")
+                    .email("person.2@example.com")
+                    .role("Test Person")
+                    .build()
+            ))
+            .build())
+        .declarations(DeclarationsDto.builder()
+            .informationIsCorrect(true)
+            .discussedWithTpd(false)
+            .notGuaranteed(true)
+            .build())
+        .build();
+
+    byte[] generatedBytes = service.generatePdf(dto, "admin");
+
+    int problems = compareGeneratedPdf("ltft-admin", generatedBytes);
+    assertThat("Unexpected PDF comparison problem count.", problems, is(0));
+  }
+
+  /**
+   * Compare the bytes of a generated PDF against an existing example.
+   *
+   * @param resource          The filename of the existing PDF to check against, without extension.
+   * @param generatedPdfBytes The generated bytes.
+   * @return The number of discrepancies between the generation and expected results.
+   * @throws IOException If output files could not be created, or reading the bytes fails.
+   */
+  private int compareGeneratedPdf(String resource, byte[] generatedPdfBytes) throws IOException {
```

Comment:
🌮 nice to have the generated files to compare by hand/eye if there are differences